lib/rubygems/local_remote_options.rb

```diff
@@ -24,8 +24,10 @@ def accept_uri_http
         raise OptionParser::InvalidArgument, value
       end
 
-      unless ['http', 'https', 'file', 's3'].include?(uri.scheme)
-         raise OptionParser::InvalidArgument, value
+      allowed_schemes = ['http', 'https', 'file', 's3']
+      unless allowed_schemes.include?(uri.scheme)
+        msg =  "#{value}\nPreface URLs with one of #{allowed_schemes}"
+        raise OptionParser::InvalidArgument, msg
```

Comment:
@segiddins It looks like I was wrong, and my fix isn't working. Upon further testing, it looks like any message that's passed on line 30 when ```OptionParser::InvalidArgument``` is raised is completely discarded, and replaced with whatever ```value``` is, since it's the iterator in the block.

I'm currently experimenting with trying to get any custom error message to be returned.